src/rpc/blockchain.cpp

```diff
@@ -1168,7 +1170,7 @@ UniValue getblockchaininfo(const JSONRPCRequest& request)
             + HelpExampleRpc("getblockchaininfo", "")
         );
 
-    LOCK(cs_main);
+    LOCK2(cs_main, cs_LastBlockFile); // cs_LastBlockFile for CalculateCurrentUsage()
```

Comment:
As @sipa already commented, `CalculateCurrentUsage()` should be responsible for concurrency locking (not the calling code part).